osimpipeline/vital_tasks.py

```diff
@@ -986,109 +1162,342 @@ def __init__(self, trial, mrs_setup_task, **kwargs):
+            **kwargs)
+        self.mrs_setup_task = mrsmod_task.mrs_setup_task
+        self.doc = 'Postprocess modified DeGroote Muscle Redundancy Solver problem in MATLAB.'
+        self.mrsmod_task = mrsmod_task
+        self.path = self.mrsmod_task.path
+        self.id = self.mrs_setup_task.tricycle.id
+        self.results_output_fpath = mrsmod_task.results_output_fpath
+
+        # Plot muscle excitations, activations, and reserve activations
+        self.add_action([self.results_output_fpath],
+                        [os.path.join(self.path, 'muscle_activity.pdf'),
+                        os.path.join(self.path, 'reserve_activity.pdf')],
+                        self.plot_activations
+                        )
+
+        # Plot joint moment breakdown
+        self.add_action([self.results_output_fpath],
+                        [os.path.join(self.path, 'joint_moment_breakdown.pdf'),
+                        os.path.join(self.path, '%s_%s_mrs_moments.csv' %
+                        (self.study.name, self.id))],
+                        self.plot_joint_moment_breakdown
+                        )
+
+    def plot_activations(self, file_dep, target):
+
+        # Load mat file fields
+        muscle_names = util.hdf2list(file_dep[0], 'MuscleNames',isString=True)
+        exc = util.hdf2pandas(file_dep[0], 'MExcitation', labels=muscle_names)
+        act = util.hdf2pandas(file_dep[0], 'MActivation', labels=muscle_names)
+        dof_names = util.hdf2list(file_dep[0], 'DatStore/DOFNames', 
+            isString=True)
+        reserves = util.hdf2pandas(file_dep[0],'RActivation', labels=dof_names)
+
+        # Create plots
+        pp.plot_muscle_activity(target[0], exc=exc, act=act)
+        pp.plot_reserve_activity(target[1], reserves)
+
+    def plot_joint_moment_breakdown(self, file_dep, target):
+
+        # Load mat file fields
+        muscle_names = util.hdf2list(file_dep[0], 'MuscleNames', isString=True)
+        dof_names = util.hdf2list(file_dep[0],'DatStore/DOFNames',isString=True)
+        num_dofs = len(dof_names)
+        num_muscles = len(muscle_names)
+        joint_moments_exp = util.hdf2numpy(file_dep[0], 'DatStore/T_exp')
+        tendon_forces = util.hdf2numpy(file_dep[0], 'TForce')
+        exp_time = util.hdf2numpy(file_dep[0], 'DatStore/time').transpose()[0]
+        time = util.hdf2numpy(file_dep[0], 'Time').transpose()[0]
+        moment_arms_exp = util.hdf2numpy(file_dep[0], 'DatStore/dM').transpose()
+        
+        # Clip large tendon forces at final time
+        from warnings import warn
+        for imusc in range(len(muscle_names)):
+            tendon_force = tendon_forces[:,imusc]
+            if (tendon_force[-1] > 5*tendon_force[-2]):
+                tendon_force[-1] = tendon_force[-2]
+                tendon_forces[:,imusc] = tendon_force
+                warn('WARNING: large %s tendon force at final time. '
+                    'Clipping...' % muscle_names[imusc])
```

Comment:
Is there a reason that the check is only at the final time? Might be good to document.

Reply:
This is mostly for a workaround in my project, where I was getting large spikes at the end of the gait cycle so plotting results we're often skewed. I'll probably change this to an option, not a default setting. Probably a good reason for having a separate MRS module, for things specific to that framework like this.